app/src/ui/diff/diff-helpers.tsx

```diff
@@ -352,3 +353,52 @@ export function canSelect(
 ): file is WorkingDirectoryFileChange {
   return file instanceof WorkingDirectoryFileChange
 }
+
+/** Gets the width in pixels of the diff line number gutter based on the number of digits in the number */
+export function getLineWidthFromDigitCount(digitAmount: number): number {
+  let diffSize: number
+
+  if (digitAmount <= 3) {
+    diffSize = 35
+  } else {
+    diffSize = 10 * digitAmount + 5
+  }
+
+  return diffSize
+}
+
+/** Utility function for getting the digit count of the largest line number in an array of diff hunks */
+export function getLargestLineNumber(hunks: DiffHunk[]): number {
+  let isValidLine = false
+  let currentHunkIndex = hunks.length - 1
+  let currentLineIndex = hunks[currentHunkIndex].lines.length - 1
```

Comment:
Shouldn't we have checks for the length of those arrays?
```suggestion
  if (hunks.length === 0) {
    return 0
  }
  
  let currentHunkIndex = hunks.length - 1
  
  if (hunks[currentHunkIndex].lines.length === 0) {
    return 0
  }
  
  let currentLineIndex = hunks[currentHunkIndex].lines.length - 1
```

Reply:
I removed most of this since the below logic does away with a lot of this anyway.